pkgs/applications/graphics/tesseract/default.nix

```diff
@@ -1,67 +1,83 @@
-# Supported list of languages or `null' for all available languages
+# List of languages like [ "eng" "spa" ... ] or `null` for all available languages
 , enableLanguages ? null
-# if you want just a specific list of languages, optionally specify a hash
-# to make tessdata a fixed output derivation.
-, enableLanguagesHash ? (if enableLanguages == null # all languages
-                         then "1h48xfzabhn0ldbx5ib67cp9607pr0zpblsy8z6fs4knn0zznfnw"
-                         else null)
+
+, tesseractBase ? stdenv.mkDerivation rec {
+    name = "tesseract-${version}";
+    version = "3.05.00";
+
+    src = fetchFromGitHub {
+      owner = "tesseract-ocr";
+      repo = "tesseract";
+      rev = version;
+      sha256 = "11wrpcfl118wxsv2c3w2scznwb48c4547qml42s2bpdz079g8y30";
+    };
+
+    enableParallelBuilding = true;
+
+    nativeBuildInputs = [ pkgconfig autoreconfHook ];
+    buildInputs = [ leptonica libpng libtiff icu pango opencl-headers ];
+
+    LIBLEPT_HEADERSDIR = "${leptonica}/include";
+
+    meta = {
+      description = "OCR engine";
+      homepage = https://github.com/tesseract-ocr/tesseract;
+      license = stdenv.lib.licenses.asl20;
+      maintainers = with stdenv.lib.maintainers; [ viric earvstedt ];
+      platforms = with stdenv.lib.platforms; linux ++ darwin;
+    };
+  }
+
+# This argument is obsolete
+, enableLanguagesHash ? null
 }:
 
-let tessdata = stdenv.mkDerivation ({
-  name = "tessdata";
-  src = fetchFromGitHub {
-    owner = "tesseract-ocr";
-    repo = "tessdata";
-    rev = "3cf1e2df1fe1d1da29295c9ef0983796c7958b7d";
-    # when updating don't forget to update the default value fo enableLanguagesHash
-    sha256 = "1v4b63v5nzcxr2y3635r19l7lj5smjmc9vfk0wmxlryxncb4vpg7";
-  };
-  buildCommand = ''
-    cd $src;
-    for lang in ${if enableLanguages==null then "*.traineddata" else stdenv.lib.concatMapStringsSep " " (x: x+".traineddata") enableLanguages} ; do
-      install -Dt $out/share/tessdata $src/$lang ;
-    done;
-  '';
-  preferLocalBuild = true;
-  } // (stdenv.lib.optionalAttrs (enableLanguagesHash != null) {
-  # when a hash is given, we make this a fixed output derivation.
-  outputHashMode = "recursive";
-  outputHashAlgo = "sha256";
-  outputHash = enableLanguagesHash;
-  }));
-in
+let
+  languages = (import ./languages.nix { inherit stdenv fetchurl fetchFromGitHub; }).v3;
 
-stdenv.mkDerivation rec {
-  name = "tesseract-${version}";
-  version = "3.05.00";
+  tessdata = if enableLanguages == null then
+      languages.all
+    else
+      map (lang: languages.${lang}) enableLanguages;
 
-  src = fetchFromGitHub {
-    owner = "tesseract-ocr";
-    repo = "tesseract";
-    rev = version;
-    sha256 = "11wrpcfl118wxsv2c3w2scznwb48c4547qml42s2bpdz079g8y30";
-  };
+  tesseractWithData = tesseractBase.overrideAttrs (_: {
+    inherit tesseractBase;
 
-  enableParallelBuilding = true;
+    # tessdata can be a list of files or a directory containing files
+    inherit tessdata;
```

Comment:
How does this work without adding `tesseractBase` and `tessdata` to `passthru`?

Reply:
`tesseractBase` is [passed thru](https://github.com/NixOS/nixpkgs/blob/c67bb4ac9c18602ae19913136cae9c8824891dd0/pkgs/applications/graphics/tesseract/default.nix#L79), but via `drv // {…}`.

You're right, `tessdata` modified via `overrideAttrs` gets lost after another `override`. In an earlier draft, `tessdata` was a main package function arg, but I deemed this style too unconvential. I've now restored that patch as it minimizes user errors. 

`tessdata` and `tesseractBase` are now still [attrs of](https://github.com/NixOS/nixpkgs/blob/c36837a80070002f1c286369be7fe794bd2904d0/pkgs/applications/graphics/tesseract/default.nix#L46) `tesseractWithData`, but purely for the nix to builder env variable conversion.